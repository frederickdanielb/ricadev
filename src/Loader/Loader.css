.loader-in-page img{
    width: fit-content;
    width: -webkit-fill-available;
}