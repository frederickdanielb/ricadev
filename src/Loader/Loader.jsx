import loadingImage from "./../../images/loader.gif";
const Loader = ({ isFullPage = true, loading = false, ...restProps }) => {
  return (
    <>
      <div className="bg-white">
        {isFullPage && loading && (
          <div className="loader-wrapper img-gif">
            <img src={loadingImage} alt="" {...restProps} />
          </div>
        )}
        {!isFullPage && loading && (
          <img src={loadingImage} alt="" {...restProps} />
        )}
      </div>
    </>
  );
};

export default Loader;
