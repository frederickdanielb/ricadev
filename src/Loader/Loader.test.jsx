import { render, fireEvent, waitFor, screen, act } from "@testing-library/react";
import Loader from ".";
import AppContextProvider from "../../context/AppContext";
import useApp from "../../hooks/useApp";

describe("loader component", () => {
    let showLoading = true;
	let container = render(
		<AppContextProvider>
			<Loader loading={showLoading} useContext={false} />
		</AppContextProvider>
	);
    it("should render correctly", async () => {
        expect(container.asFragment()).toMatchSnapshot();
    })
    it("should not render", async () => {
        showLoading = false;
        expect(container.asFragment()).toMatchSnapshot();
    })
});
