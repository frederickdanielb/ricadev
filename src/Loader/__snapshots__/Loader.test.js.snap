// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader component should not render 1`] = `<DocumentFragment />`;

exports[`loader component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="bg-white"
  />
  <div
    class="bg-white"
  >
    <img
      alt=""
      src="loader.gif"
    />
  </div>
</DocumentFragment>
`;
